app/controllers/samples_controller.rb

```diff
@@ -237,16 +237,11 @@ def report_info
     tax_ids = @report_info[:taxonomy_details][2].map { |x| x['tax_id'] }
     tax_ids |= @report_info[:taxonomy_details][2].map { |x| x['family_taxid'] }
 
-    # Get lineage records from the TaxonLineage table and get more info from TaxonCount rows.
-    # This is because taxon counts contains transformed results such as -10000543 to indicate
-    # non-specific hits.
     # TODO: Should definitely be simplified with taxonomy/lineage refactoring.
-    lineage_records = TaxonLineage.where(taxid: tax_ids)
-    lineages = TaxonCount.connection.select_all(lineage_records).to_hash
     lineage_by_taxid = {}
-    lineages.each do |x|
+    TaxonLineage.where(taxid: tax_ids).as_json.each { |x|
       lineage_by_taxid[x['taxid']] = x
-    end
+    }
```

Comment:
beautiful change

Reply:
why do you need as_json, you really dont need that. 
just do 

lineage_by_taxid[x.taxid]  = x.to_hash

Reply:
why not move 236 to 244 to fill_lineage_details?

Reply:
The TaxonLineage ActiveRecord result actually doesn't have .to_hash so as_json was basically the same thing? https://stackoverflow.com/questions/15427936/how-to-convert-activerecord-results-into-an-array-of-hashes